core/server/api/canary/oembed.js

```diff
@@ -1,8 +1,144 @@
 const common = require('../../lib/common');
-const {extract, hasProvider} = require('oembed-parser');
+const { extract, hasProvider } = require('oembed-parser');
 const Promise = require('bluebird');
 const request = require('../../lib/request');
 const cheerio = require('cheerio');
+const domino = require('domino');
+const {getMetadata} = require('page-metadata-parser');
+
+function generateBookmarkHtml(data)  {
+    const html = `
+    <style>
+    figure {
+        margin: 0;
+        padding: 0;
+        font-family: sans-serif;
+      }
+      
+      .kg-bookmark-card {
+        border: 1px solid #efefef;
+        background: #fafafa;
+      }
+      
+      .kg-bookmark-container a {
+        display: flex;
+        color: #222;
+        text-decoration: none;
+      }
+      
+      .kg-bookmark-content {
+        display: flex;
+        flex-wrap: wrap;
+        flex-basis: 66%;
+        align-items: start;
+        align-content: flex-start;
+        padding: 20px;
+      }
+      
+      .kg-bookmark-title {
+        flex-grow: 1;
+        font-size: 1.2em;
+        font-weight: 600;
+      }
+      
+      .kg-bookmark-container a:hover .kg-bookmark-title {
+        color: #3EB0EF;
+      }
+      
+      .kg-bookmark-description {
+        line-height: 1.4em;
+        margin-top: 12px;
+      }
+      
+      .kg-bookmark-thumbnail {
+        flex-basis: 34%;
+        max-height: 180px;
+      }
+      
+      .kg-bookmark-thumbnail img {
+        width: 100%;
+        height: 100%;
+        object-fit: cover;
+      }
+      
+      .kg-bookmark-metadata {
+        display: flex;
+        align-items: center;
+        font-size: 0.8em;
+        color: #777;
+        margin-top: 12px;
+      }
+      
+      .kg-bookmark-logo {
+        width: 18px;
+        height: 18px;
+        margin-right: 6px;
+      }
+      
+      .kg-bookmark-author:after {
+        content: "•";
+        margin: 0 6px;
+      }
+      </style>
+    
+    <figure class="kg-card kg-bookmark-card">
+  <div class="kg-bookmark-container">
+    <a href="${data.url}">
+      <div class="kg-bookmark-content">
+        <div class="kg-bookmark-title">${escape_HTML(data.title)}</div>
+        <div class="kg-bookmark-description">${escape_HTML(data.description)}</div>
+        <div class="kg-bookmark-metadata">
+          <img src="${data.icon}" class="kg-bookmark-logo">
+          <span class="kg-bookmark-author">${data.type}</span>
+          <span class="kg-bookmark-url">${data.provider}</span>
+        </div>
+      </div>
+      <div class="kg-bookmark-thumbnail">
+        <img src="${data.image}"> </img>
+      </div>
+    </a>
+    </div>
+    </figure>
+`;
```

Comment:
@peterzimon Currently the whole html for bookmark card on admin is coming directly from here, both style and html.